sdk/go/internal/types.go

```diff
@@ -635,6 +635,26 @@ func (o *OutputState) applyTWithApplier(ctx context.Context, ap *applier) Output
 	return result
 }
 
+// DeferredOutput creates an Output whose value can be later resolved from another Output instance.
+func DeferredOutput(ctx context.Context) (Output, func(Output)) {
+	join := &WorkGroup{}
+	join.Add(1)
+	result := NewOutput(join, AnyOutputType)
+	resolve := func(o Output) {
+		go func() {
```

Comment:
Does this goroutine not need to call `join.Done()` at some point?  I also see `NewOutputState` (which is called by `NewOutput` also calls `join.Add(1)`, so maybe we just don't need to do it above instead?

Reply:
Yeah we don't need `join`, changed the implementation such that we deal with the output state directly